### PR TITLE
Support mavlink opaque ID

### PR DESCRIPTION
**Pull request by KonradRudin**
### Solved Problem
This PR adds support for the Mavlink Mission opaque ID as defined in https://github.com/mavlink/mavlink/pull/2012. It is based on the earlier implementation for the mission checksum support in https://github.com/PX4/PX4-Autopilot/pull/18418, but the mission checksum in Mavlink will be deprecated in https://github.com/mavlink/mavlink/pull/2010.

### Solution
- Add an opaque Id for the mission, geofence and rally points. The opaque id is based on a Crc32 checksum as in the implementation in https://github.com/PX4/PX4-Autopilot/pull/18418
- Opaque id is send from the System in the MISSION_COUNT on download. On upload, the opaque id is sent on the final MISSION_ACK
- The opaque Id is also streamed continuously in the MISSION_CURRENT message, such that a GCS can detect plan changes.
- The opaque id internally also replaces the mission counter, as those serves the same purpose.

### Changelog Entry
For release notes:
```
Feature: Add Support for Mavlink mission opaque Id.
```

### Alternatives
Opqaue id could also just be a counter instead of the Crc32. The hash implementation has the advantage that on loading the same mission (e.g. also from a file) should result in the same Id and thus the potentially multiple connected GCS does not need to download it again. Also it would allow for the feature, that a GCS does not need to download the mission on first connection if the old stored mission and opaque id is the same.

### Test coverage
- Unit/integration test: SITL testing by observing the mavlink message sent from the system.

### Context
- Before this can be merged, the respective mavlink PR should be merged

### CODE REVIEW

File: .gitmodules
Comment (ThomasDebrunner) [inline .gitmodules:3]: I assume this will have to be changed to upstream mavlink, once it is merged in upstream mavliink, right? (Before we merge here)

File: msg/Mission.msg
Comment (ThomasDebrunner) [inline msg/Mission.msg:12]: That would now be "if changed" and not "if increased", right?

File: src/modules/mavlink/mavlink_mission.cpp
Comment (ThomasDebrunner) [inline src/modules/mavlink/mavlink_mission.cpp:527]: Why is this needed?

File: src/modules/mavlink/mavlink_mission.cpp
Comment (ThomasDebrunner) [inline src/modules/mavlink/mavlink_mission.cpp:412]: I misunderstood this as some sort of CRC for the current mission item (e.g. waypoint or something). Maybe rename to "current_mission_type_crc" or similar?

File: .gitmodules
Comment (KonradRudin) [reply .gitmodules:3]: Exactly, this PR will stay in draft because of this. I need to update this as soon as it is merged on Mavlink

File: src/modules/mavlink/mavlink_mission.cpp
Comment (KonradRudin) [reply src/modules/mavlink/mavlink_mission.cpp:527]: The mission current has now the IDs which should be periodically send. That means it should send it periodically as well, when there is no mission available, so that the ground station is informed when the mission is cleared.

**Review by ThomasDebrunner — DISMISSED** [review]

**Comment by hamishwillee** [comment]
@KonradRudin Looks like this is getting close. Let me know when it is ready to merge and you have test results/stuff we can use to validate this.

**Review by dagar — APPROVED** [review]

**Comment by dagar** [comment]
As soon as https://github.com/mavlink/mavlink/pull/2012 is merged we can update the mavlink submodule and bring this in.

**Comment by KonradRudin** [comment]
rebased on main again

**Comment by KonradRudin** [comment]
@hamishwillee This PR was discussed now in the maintainer call and in agreement wirh @dagar . This side is ready now

**Comment by KonradRudin** [comment]
Upstream mavlink is merged now. Update the mavlink submodule

**Review by dagar — APPROVED** [review]

**Pull request merged**
